orchestra/executor.py

```diff
@@ -348,8 +348,17 @@ def _try_group_orders(dependency_graph, group):
             for g1, g2 in zip(permutation, permutation[1:]):
                 # Add edge from all nodes in g1 to all nodes in g2
                 for a1, a2 in product(g1, g2):
-                    if a1 is not a2:
-                        depgraph_copy.add_edge(a1, a2)
+                    same_action = a1 is a2
+                    same_build = (
+                            isinstance(a1, ActionForBuild)
+                            and isinstance(a2, ActionForBuild)
+                            and a1.build is a2.build
```

Comment:
How is this different from just `a1.build is a2.build`?

Reply:
If `a{1,2}` are `ActionForComponent` they won't have a `.build` property